### ISSUE TITLE
Sending invoice to a non-existing account sometimes sends it to a wrong email

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#48606]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.29-6
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @CyberAndrii 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1725470168205369?thread_ts=1725439143.231259&cid=C01GTK53T8Q

## Action Performed:
1. Open new.expensify.com and log in
2. Create a new workspace
3. Send an invoice to an email that is not yet registered
4. You may need to repeat step 3 multiple times

## Expected Result:
The invoice is sent to the specified email.
## Actual Result:
The invoice is sent to an unfamiliar email address.
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/2df0d082-031d-42ec-9188-92210059f246

https://github.com/user-attachments/assets/6b864436-fb11-4dab-a112-b914659ab3ae

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by sakluger** [comment]
This is super weird. This feels like it may be a BE issue, but I'll add the External label to start so we can get some eyes on it.

**Comment by ahmedGaber93** [comment]
Not overdue, No proposals yet.

**Comment by sakluger** [comment]
@ahmedGaber93 do you have any sense if this is an external or BE issue?

**Comment by ahmedGaber93** [comment]
@sakluger Yes, I think that. I see the report action `invited @user to chat` and the personal details for the non-exist user is created by the backend side and not set optimistically, and the frontend sending the related parameter `receiverEmail: weweweweerr12@wewe.com` correctly to the `api/SendInvoice`, and  the wrong data come in `api/openReport` when opening the admin room report.

**Comment by ChavdaSachin** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- Approver to submit on behalf owner produces error

### What is the root cause of that problem?
- We are not handling the approver case https://github.com/Expensify/App/blob/498d45cc55bae4dde9f85036c75d5a421e0469e9/src/libs/actions/IOU.ts#L7161-L7176

### What changes do you think we should make in order to solve the problem?
- Add additional case for approver 
```js
    const approverAccountID = policy?.role === CONST.POLICY.ROLE.AUDITOR ? currentUserPersonalDetails?.accountID : undefined;
```
and pass this here
```js
    const optimisticSubmittedReportAction = ReportUtils.buildOptimisticSubmittedReportAction(expenseReport?.total ?? 0, expenseReport.currency ?? '', expenseReport.reportID, adminAccountID ?? approverAccountID);
```
https://github.com/Expensify/App/blob/498d45cc55bae4dde9f85036c75d5a421e0469e9/src/libs/actions/IOU.ts#L7173
And do some variable renaming afterwards.
### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by ahmedGaber93** [comment]
@sakluger bump https://github.com/Expensify/App/issues/48606#issuecomment-2353600846

**Comment by sakluger** [comment]
@ahmedGaber93 thanks for responding. Just to confirm, you are saying this is a BE issue? I couldn't tell from you response.

**Comment by ahmedGaber93** [comment]
@sakluger Yes, this is a BE issue.

**Comment by ahmedGaber93** [comment]
@sakluger this is a BE issue. What's next?

**Comment by sakluger** [comment]
I asked for volunteers in Slack: https://expensify.slack.com/archives/CSL3XBCCR/p1726761957350089

**Comment by sakluger** [comment]
Bumped in Slack.

**Comment by sakluger** [comment]
Thanks @madmax330 for taking a look!

**Comment by ahmedGaber93** [comment]
The issue has just moved to the internal, and we are waiting for the investigation result

**Comment by sakluger** [comment]
@madmax330 have you had a chance to look at this one?

**Comment by madmax330** [comment]
No not yet, I will look tomorrow since I'm headed out for today

**Comment by sakluger** [comment]
@madmax330 any updates?

**Comment by madmax330** [comment]
i looked in the code, but I didn't find anything obvious. I will have to try to reproduce it locally to see what's happening

**Comment by sakluger** [comment]
Hey @madmax330 have you had any more time to look into this one?

**Comment by sakluger** [comment]
@madmax330 any updates?

**Comment by madmax330** [comment]
Still not a priority. I double checked, and it's not actually sending the invoice to non-existing accounts, it's adding accounts to the admins room. I'm not sure why.

**Comment by sakluger** [comment]
I assume this is still not a priority. I do think we should fix this one eventually.

**Comment by sakluger** [comment]
No updates.

**Comment by sakluger** [comment]
No updates.

**Comment by sakluger** [comment]
No updates.

**Comment by madmax330** [comment]
No update

**Comment by sakluger** [comment]
Still no updates.

**Comment by sakluger** [comment]
No updates. @madmax330 is this prioritized correctly? Or should we maybe find someone else to help? I think this is an important thing to fix, but it's been open now for over 4 months.

**Comment by tylerkaraszewski** [comment]
The PR for this was deployed. This didn't auto-close because it was an Auth change. I'll close it.
